CHANGELOG.md

```diff
@@ -1,3 +1,7 @@
+### v1.17.1 - 2024/09/17
```

Comment:
This version should match the one inside `share/version.py`:

https://github.com/elastic/elastic-serverless-forwarder/blob/8be4fc4b87a06aa857e0cbc227f55f0533d89035/share/version.py#L5

Perhaps we should update the workflow to compare these, this should not pass...

Reply:
But updating the `version.py` will also trigger a new release, which seems a bit unnecessary, since the dependencies are not changing (the dependencies can be seen [here](https://github.com/elastic/elastic-serverless-forwarder/blob/8be4fc4b87a06aa857e0cbc227f55f0533d89035/.github/workflows/upload-dependencies.yml#L55-L59)).

Reply:
Yeah, @constanca-m, you're right. 

@hvltaj, the `publish_lambda.sh` is not part of the release; it's a supporting tool, so we can avoid updating the `CHANGELOG.md` and `share/version.py` files.

Reply:
I updated the changelog after https://github.com/elastic/elastic-serverless-forwarder/pull/803 is merged. Now it should match the version inside share/version.py.

Reply:
ahh good point, ok I will take out the entry from CHANGELOG.md then!

Reply:
I have created https://github.com/elastic/elastic-serverless-forwarder/issues/807 to see how we can stop this from happening, or maybe ease this version update process a bit.